wordpress-mobile/WordPress-iOS
# Fix screenshot test for Russian & Thai

Fixes #15074

As noted in the issue, for some reason typing text works in most text fields and languages _except_ secure text entry fields in Russian and Thai (of the languages we run the test in).

That issue prevented the test from logging in when the device is set to those languages. To work around that, this PR checks the language code on the device and pastes (instead of typing) the password in those languages. I didn't switch to using a paste method for all languages because in repeated test runs I found pasting to be a bit flaky, so this limits it to the languages where it's really needed.

We also have a small mocks improvement in https://github.com/wordpress-mobile/WordPressMocks/pull/21 so I've updated the Podfile here so we can test that change.

To test:

* Run `rake mocks`.
* Set your simulator to Russian and Thai and confirm the WordPressScreenshotGeneration test passes.
* Confirm the WordPressScreenshotGeneration test continues to pass in another language (e.g. English).

(You can run the test at least once on an iPad Pro, 12.9 inch, 2nd or 3rd gen to test the mocks change.)

PR submission checklist:
